myapp/views.py

```diff
@@ -1,23 +1,32 @@
 from django.shortcuts import render, redirect
 from myapp import stock_api
-from myapp.models import Stock
+from myapp.models import Stock, Profile
 from django.http import JsonResponse
 from django.contrib.auth.models import User
-from django.contrib.auth import logout
+from django.views.decorators.http import require_http_methods
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth import login, logout
 
 
 # View for the home page - a list of 20 of the most active stocks
 def index(request):
 	# Query the stock table, filter for top ranked stocks and order by their rank.
 	data = Stock.objects.filter(top_rank__isnull=False).order_by('top_rank')
-	return render(request, 'index.html', {'page_title': 'Main', 'data': data })
+	profile = None
+	if request.user.is_authenticated:
+		profile = Profile.objects.get(id=request.user.id)
+	return render(request, 'index.html', {'page_title': 'Main', 'data': data, 'profile': profile})
 
 
 # View for the single stock page
 # symbol is the requested stock's symbol ('AAPL' for Apple)
 def single_stock(request, symbol):
 	data = stock_api.get_stock_info(symbol)
-	return render(request, 'single_stock.html', {'page_title': 'Stock Page - %s' % symbol, 'data': data})
+	stock = Stock.objects.get(symbol=symbol)

```

Comment:
I hope there won't be any merging conflicts in this section of the code, becuase this view function has changed a lot with error handling that @murradkh added